src/main/java/su/litvak/chromecast/api/v2/ChromeCast.java

```diff
@@ -86,6 +86,21 @@ public String getApplication() {
     public void setApplication(String application) {
         this.application = application;
     }
+    
+    /** Returns the {@link #channel} and opens it if it's not yet or
+     * no longer open.
+     * @return an open channel.
+     */
+    private Channel channel() throws IOException {
```

Comment:
I got your point. Thank you for the explanation too. 

I am not sure what will be the best solution here, maybe I should just update the JavaDoc for getters and all public methods with a reference to `connect()` method. I personally like to let user control the connection to chromecast device himself. This way he can do some short-sessions, like for every action, or long-session if needed.